o2a/transformers/base_transformer.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Base transformer classes"""
+from abc import ABC, abstractmethod
+from typing import Type
+
+from o2a.converter.parsed_action_node import ParsedActionNode
+from o2a.converter.workflow import Workflow
+
+
+# pylint: disable=too-few-public-methods
+class BaseWorkflowTransformer(ABC):
+    """
+    Base class for all transformers
+    """
+
+    @abstractmethod
+    def process_workflow(self, workflow: Workflow):
+        pass
+
+
+class TypeNodeWorkflowTransformer(BaseWorkflowTransformer):
```

Comment:
I think we should move all the methods to BaseWorkflowTransformer. I think "process_workflow" should always look the same.

Reply:
That's not true. There are cases when process_workflow can look different.
I created a separate PR, which shows it.
https://github.com/GoogleCloudPlatform/oozie-to-airflow/pull/260/commits/60ff6d1589d02a21122771de302a9ac30b42c970
This is not part of this PR, because in this PR I tried not to introduce new functionalities.

Reply:
This mapper only processes nodes in a sequential manner.  This is enough in many cases and that's why I created this class. The concept of transformers allows you to operate on more complex structures.

Reply:
I understand that (and I looked at 60ff6d1 before making this comment). The 60ff6d1 is also sequentially processing - but "all nodes" - and I think we can easily change 60ff6d1 to get the list of "all types" (see the other comment). Also you can always override process_workflow() and do not use super().process_workflow() at all. I prefer to have as small hierarchy of classes and avoid fully abstract classes (and this case seems YAGNI).

Reply:
I removed this classes.